test/spec/factor-enroll.js

```diff
@@ -0,0 +1,109 @@
+var util = require('../util/util'),
+    _ = require('lodash');
+
+describe('FACTOR_ENROLL', function () {
```

Comment:
For now I have only added tests for `FACTOR_ENROLL ` since that is closely tied to my change. But i saw that there are tests for other states. I can create a follow up JIRA to beef up the tests support for new IDX pipeline states `FACTOR_REQUIRED`, `FACTOR_CHALLENGE`, `PROFILE_REQUIRED` and `FACTOR_ENROLL_ACTIVATE`